### ISSUE TITLE
Rename 'flux' column in tables generated by ``StarFinderBase`` objects

### ISSUE THREAD

**Issue by mirca** [astropy/photutils#519]
I think it makes sense to rename the ``flux`` column in tables generated by ``StarFinderBase`` objects to something like ``flux_peak_density``. I'm saying this because the name ``flux`` induces users to use those numbers as the initial guesses for the flux parameter in the PSF photometry code.

P.S.: I myself used ``flux`` from those tables as initial guesses for PSF photometry which end up generating bad results because those values were too far from the "true" flux value, thanks @eteq saw that and we end up using aperture photometry to generate quite good guesses.

**Comment by larrybradley** [comment]
@mirca Yes, I agree with that this will limit confusion.  However, this change should apply only to `DAOStarFinder`.  The output ``flux`` from `IRAFStarFinder` actually is a background-subtracted flux (I think the original `IRAFStarFinder` docs got overwritten when we moved them to the `StarFinderBase` class).  In other words, the `find_stars` documentation needs be different for the two finders.  I can work on this since there are other things I want to clean up in the star finders.
